Add description about placeholders of database.properties (#1498)
scalar-labs/scalardb: docs/configurations.md

| `scalar.db.default_namespace_name`                               | The given namespace name will be used by operations that do not already specify a namespace.                                                                                                |                      |
| `scalar.db.system_namespace_name`                                | The given namespace name will be used by ScalarDB internally.                                                                                                                               | `scalardb`           |

> ## Use placeholders

Reviewer: For consistency, let's change this to a noun phrase as well.

```suggestion
## Placeholder usage
```

> You can use placeholders in the values, and they are replaced with environment variables (${env:<ENVIRONMENT_VARIABLE_NAME>}) or system properties (${sys:<SYSTEM_PROPERTY_NAME>}). You can also specify default values in placeholders like ${sys:<SYSTEM_PROPERTY_NAME>:-<DEFAULT_VALUE>}.

Reviewer 2: > environment variables (${env:<ENVIRONMENT_VARIABLE_NAME>})

How about syntax-highlighting it as follows?

```md
environment variables (`${env:<ENVIRONMENT_VARIABLE_NAME>}`)
```

Author: Ah, I see. Looks good!
I will apply it. Thank you for your suggestion!

```

For details about client configurations, see the ScalarDB Cluster [client configurations (redirects to the Enterprise docs site)](https://scalardb.scalar-labs.com/docs/latest/scalardb-cluster/developer-guide-for-scalardb-cluster-with-java-api/#client-configurations).

> ### Configuration example #3 - Use placeholders

Reviewer: Revised this heading to be a noun phrase since the other examples follow a noun phrase pattern.

```suggestion
### Configuration example #3 - Placeholders usage
```

> ### Configuration example #3 - Placeholder usage

Reviewer 3: I think we can move this example into the `## Placeholder usage` part. I think this `Configuration example` section is more like for the actual/real settings. So if this example is only for the placeholder usage, I think we can put it in the the `## Placeholder usage` part. What do you think?

Author: I see. Indeed, this example does not include the actual configuration (minimum configuration) to access the backend DB or ScalarDB Cluster.
I agree with you. I will move this example into the `## Placeholder usage` section. Thank you!

Author: I updated it in 910c2693c0d537f169da70c4504824ea45387fba.

> ### Configuration example #3 - Placeholders usage

Reviewer: My mistake. `Placeholders` needs to be singular here.

```suggestion
### Configuration example #3 - Placeholder usage
```

> The following is an example of the configuration using placeholders:

Reviewer: ```suggestion
The following is an example of a configuration that uses placeholders:
```

```properties
scalar.db.username=${env:<SCALAR_DB_USERNAME>:-admin}
scalar.db.password=${env:<SCALAR_DB_PASSWORD>}
```

> In this example configuration, ScalarDB reads username and password from environment variables. If the environment variable `SCALAR_DB_USERNAME` does not exist, ScalarDB uses the default value `admin`.

Reviewer: ```suggestion
In this example configuration, ScalarDB reads the username and password from environment variables. If the environment variable `SCALAR_DB_USERNAME` does not exist, ScalarDB uses the default value `admin`.
```
